client/src/main/java/io/pravega/client/state/SynchronizerConfig.java

```diff
@@ -41,8 +51,47 @@
      */
     int readBufferSize;
     
-    public static class SynchronizerConfigBuilder {
+    public static class SynchronizerConfigBuilder implements ObjectBuilder<SynchronizerConfig> {
         private EventWriterConfig eventWriterConfig = EventWriterConfig.builder().retryAttempts(Integer.MAX_VALUE).enableConnectionPooling(true).build();
         private int readBufferSize = 256 * 1024;
     }
+
+    private static class SynchronizerConfigSerializer
+            extends VersionedSerializer.WithBuilder<SynchronizerConfig, SynchronizerConfigBuilder> {
+        @Override
+        protected SynchronizerConfigBuilder newBuilder() {
+            return builder();
+        }
+
+        @Override
+        protected byte getWriteVersion() {
+            return 0;
+        }
+
+        @Override
+        protected void declareVersions() {
+            version(0).revision(0, this::write00, this::read00);
+        }
+
+        private void read00(RevisionDataInput revisionDataInput, SynchronizerConfigBuilder builder) throws IOException {
+            builder.readBufferSize(revisionDataInput.readInt());
+            builder.eventWriterConfig.fromBytes(ByteBuffer.wrap(revisionDataInput.readArray()));
+        }
+
+        private void write00(SynchronizerConfig object, RevisionDataOutput revisionDataOutput) throws IOException {
+            revisionDataOutput.writeInt(object.getReadBufferSize());
+            revisionDataOutput.writeArray(object.eventWriterConfig.toBytes().array());
```

Comment:
You need to provide arrayOffset and length as it is legal for a byteBuffer's array to have more data in in that just the buffer. (And this as calling code should not depend on anything about the implementation of eventWriter's toBytes method that is not in the signature)